compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -762,47 +636,54 @@ pub fn type_metadata<'ll, 'tcx>(cx: &CodegenCx<'ll, 'tcx>, t: Ty<'tcx>) -> &'ll
     };
 
     {
-        let mut type_map = debug_context(cx).type_map.borrow_mut();
-
         if already_stored_in_typemap {
-            // Also make sure that we already have a `TypeMap` entry for the unique type ID.
-            let Some(metadata_for_uid) = type_map.find_metadata_for_unique_id(unique_type_id) else {
-                bug!(
-                    "expected type metadata for unique \
-                            type ID '{}' to already be in \
-                            the `debuginfo::TypeMap` but it \
-                            was not. (Ty = {})",
-                    type_map.get_unique_type_id_as_string(unique_type_id),
-                    t
-                );
-            };
-
-            match type_map.find_metadata_for_type(t) {
-                Some(metadata) => {
-                    if metadata != metadata_for_uid {
+            // Make sure that we really do have a `TypeMap` entry for the unique type ID.
+            let metadata_for_uid =
+                match debug_context(cx).type_map.find_metadata_for_unique_id(unique_type_id) {
+                    Some(metadata) => metadata,
+                    None => {
                         bug!(
-                            "mismatch between `Ty` and \
-                                   `UniqueTypeId` maps in \
-                                   `debuginfo::TypeMap`. \
-                                   UniqueTypeId={}, Ty={}",
-                            type_map.get_unique_type_id_as_string(unique_type_id),
+                            "expected type metadata for unique \
+                               type ID '{:?}' to already be in \
+                               the `debuginfo::TypeMap` but it \
+                               was not. (Ty = {})",
+                            unique_type_id,
```

Comment:
I think effectively we're going to get `t` printed twice in this message since `UniqueTypeId` now directly wraps a `Ty`. Perhaps we should call `unique_type_id.to_string()` here to get the fingerprint (if that's useful)? Otherwise, we probably only need to include the `Ty` once.

Reply:
I just removed it. The important info for debugging is knowing which type the error occurred for.